### PR TITLE
Detail Screen - API for similar photos 

### PR DESCRIPTION
**Pull request by at-huyvo2**
- API for similar photos.
- Custom header for comments section

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/Library/Ext/StringExt.swift [hunk 1]
```diff
@@ -41,4 +41,9 @@ extension String {
     func stripHtml() -> String {
        return self.replacingOccurrences(of: "<[^>]+>", with: "", options: .regularExpression, range: nil)
     }
+
+    func getFirstHastag() -> String {
+        let hashtagArray = self.components(separatedBy: " ")
+        return hashtagArray[0]
```
Comment (at-hainguyen3) [inline MyApp/Library/Ext/StringExt.swift]: Chỗ này dễ bị crash app

File: MyApp/View/Controller/DetailViewController/DetaiViewModel.swift [hunk 2]
```diff
@@ -17,7 +17,7 @@ final class DetailViewModel: ViewModel {
     var user: User?
     var secctionTypes: [SecctionType] = [.comment]
     var comments: [Comment] = []
-
+    var feeds: [Feed] = []
     enum Result {
```
Comment (at-hainguyen3) [inline MyApp/View/Controller/DetailViewController/DetaiViewModel.swift]: Bỏ enum này trong struct Config luôn

File: MyApp/View/Controller/DetailViewController/DetailViewController.swift [hunk 3]
```diff
@@ -20,12 +21,15 @@ final class DetailViewController: BaseViewController {
...
     @IBOutlet private weak var sendButtonView: UIView!
+    @IBOutlet private weak var collectionView: UICollectionView!
+    @IBOutlet private weak var footerView: UIView!
 
     // MARK: Properties
     var viewModel = DetailViewModel()
     let dispatchGroup = DispatchGroup()
     var getUserResult = false
     var getCommemtResult = false
+    var getSimilarResult = false
```
Comment (at-hainguyen3) [inline MyApp/View/Controller/DetailViewController/DetailViewController.swift]: Đặt 3 biến này bên view model

File: MyApp/Model/API/Core/Api.swift [hunk 4]
```diff
@@ -19,6 +19,7 @@ final class Api {
     struct Feed { }
     struct User { }
     struct Comment { }
+    struct PhotosSimilar { }
```
Comment (at-daiho) [inline MyApp/Model/API/Core/Api.swift]: ERROR - Type name should start with an uppercase character: 'photosSimilar'

File: MyApp/Model/API/Core/Api.swift [hunk 5]
```diff
@@ -19,6 +19,7 @@ final class Api {
     struct Feed { }
     struct User { }
     struct Comment { }
+    struct photosSimilar { }
```
Comment (at-daiho) [inline MyApp/Model/API/Core/Api.swift]: ERROR - Type name should start with an uppercase character: 'photosSimilar'

File: MyApp/Model/API/Core/Api.swift [hunk 6]
```diff
@@ -19,6 +19,7 @@ final class Api {
     struct Feed { }
     struct User { }
     struct Comment { }
+    struct Similar { }
```
Comment (at-daiho) [inline MyApp/Model/API/Core/Api.swift]: ERROR - Type name should start with an uppercase character: 'photosSimilar'

File: MyApp/View/Controller/DetailViewController/DetaiViewModel.swift [hunk 7]
```diff
@@ -101,12 +118,36 @@ extension DetailViewModel {
...
+        let size = CGSize(width: width, height: height)
+        return size
+    }
 }
 
 extension DetailViewModel {
 
     struct Config {
+
+        enum Result {
```
Comment (at-daiho) [inline MyApp/View/Controller/DetailViewController/DetaiViewModel.swift]: WARNING - Types should be nested at most 1 level deep

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

**Comment by at-hainguyen3** [comment]
LGTM

**Pull request merged**